jpa/subsystem/src/main/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParser.java

```diff
@@ -49,6 +49,19 @@
 public class PersistenceUnitXmlParser extends MetaDataElementParser {
     // cache the trace enabled flag
     private static final boolean traceEnabled = ROOT_LOGGER.isTraceEnabled();
+    private static final Version DEFAULT_VERSION;
+
+    static {
+        Version defaultVersion;
+        try {
+            // Try and load a jakarta namespace JPA API class to see if we're EE 8 or a later EE
+            PersistenceUnitXmlParser.class.getClassLoader().loadClass("jakarta.persistence.SharedCacheMode");
```

Comment:
Cool idea to detect the default as Jakarta EE 8 Persistence 2.2 versus EE 9 Persistence 3.0!